patches/generativeaipatch.py

```diff
@@ -0,0 +1,24 @@
+import wrapt
+import os
+
+@wrapt.when_imported('google.generativeai')
+def post_import_logic(module):
+    old_configure = module.configure
+    def new_configure(*args, **kwargs):
+        if ('default_metadata' in kwargs):
+            default_metadata = kwargs['default_metadata']
+        else:
+            default_metadata = []
+        kwargs['transport'] = 'rest' # Only support REST requests for now
+        default_metadata.append(("x-kaggle-proxy-data", os.environ['KAGGLE_DATA_PROXY_TOKEN']))
```

Comment:
KAGGLE_DATA_PROXY_TOKEN should always be set now but if we remove it in the future, this would crash. Make sure to check that this value is set and escape. This exact code may be running years from now (because people can pin their environment), it has to be bullet proof and checks all the edge cases.

Reply:
Same with `KAGGLE_USER_SECRETS_TOKEN`.